react/components/NewChannelForm/index.js

```diff
@@ -96,81 +99,131 @@ class NewChannelForm extends Component {
-              errorMessage={attributeErrors.description}
-            />
-          </LabelledInput>
-
-          <LabelledInput my={6} alignItems="start">
-            <Label>
-              Privacy
-            </Label>
-
-            <div>
-              <ChannelVisibilityPulldown
-                value={visibility.toUpperCase()}
-                onChange={this.handleVisibility}
-              />
-            </div>
-          </LabelledInput>
-
-          <LabelledInput mt={6} mb={8}>
-            <Label />
-            <LabelledCheckbox
-              onChange={this.handleVisitChannel}
-              checked={visit_channel}
-            >
-              Visit channel
-            </LabelledCheckbox>
-          </LabelledInput>
-
-          {mode === 'error' &&
-            <Text mb={6} f={2} color="state.alert" textAlign="center">
-              {errorMessage}
-            </Text>
+      <Query query={groupsCountQuery}>
+        {({ data, error, loading }) => {
+          if (loading) {
+            return (
+              <TitledDialog
+                title="New channel"
+                label={{
+                  resting: 'Create channel',
+                }[mode]}
+                onDone={this.handleSubmit}
+              >
+                <LoadingIndicator p={6} />
+              </TitledDialog>
+            );
+          }
+
+          if (error) {
+            return (
+              <TitledDialog
+                title="New channel"
+                label={{
+                  resting: 'Create channel',
+                }[mode]}
+                onDone={this.handleSubmit}
+              >
+                <Text color="state.alert" f={2} p={6}>
+                  {error.message}
+                </Text>
+              </TitledDialog>
+            );
           }
-        </div>
-      </TitledDialog>
+
+          const { me: { counts } } = data;
+
+          return (
+            <TitledDialog
+              title="New channel"
+              label={{
+                resting: 'Create channel',
+                creating: 'Creating...',
+                redirecting: 'Redirecting...',
+                success: 'Created',
+                error: 'Error',
+              }[mode]}
+              onDone={this.handleSubmit}
+              disabled={isDisabled}
+            >
+              <div>
+                <LabelledInput mt={6} mb={7}>
+                  <Label>
+                    Name
+                  </Label>
+
+                  <Input
+                    f={7}
+                    color={`channel.${visibility.toLowerCase()}`}
+                    placeholder="Type channel name"
+                    borderless
+                    autoFocus
+                    required
+                    value={title}
+                    onChange={this.handleTitle}
+                    errorMessage={attributeErrors.title}
+                  />
+                </LabelledInput>
+
+                {counts.groups > 0 &&
```

Comment:
This whole check for group existence seems overcomplicated and unnecessary. The author field can just handle this by itself and if there's no groups then it defaults to you and is disabled.